prover/crates/bin/witness_generator/src/node_aggregation.rs

```diff
@@ -97,9 +91,11 @@ impl NodeAggregationWitnessGenerator {
         skip_all,
         fields(l1_batch = %job.block_number, circuit_id = %job.circuit_id)
     )]
-    pub fn process_job_sync(
+    pub async fn process_job_impl(
```

Comment:
It would be nice to leave a comment stating something like "While this function is asynchronous, it contains blocking code, so should only be called from within `spawn_blocking` context" (same in other places).
Can be done in a follow-up, though.

Reply:
Added